From 69319cc04e07f6ac83441547eb71188f30827bb4 Mon Sep 17 00:00:00 2001
From: Ross Wille <wille@freescale.com>
Date: Thu, 6 Nov 2008 16:26:23 -0700
Subject: [PATCH] ENGR00098015 smsc911x: fix loadable module when in spi mode

Allow the smsc911x driver to be built as a loadable module
on platforms which access smsc911x ethernet via spi cpld.

Signed-off-by: Rob Herring <r.herring@freescale.com>
---
 arch/arm/mach-mx25/mx25_3stack_cpld.c |    2 ++
 arch/arm/mach-mx37/mx37_3stack_cpld.c |    2 ++
 2 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx25/mx25_3stack_cpld.c b/arch/arm/mach-mx25/mx25_3stack_cpld.c
index 5e62c77..fb36900 100644
--- a/arch/arm/mach-mx25/mx25_3stack_cpld.c
+++ b/arch/arm/mach-mx25/mx25_3stack_cpld.c
@@ -74,6 +74,7 @@ unsigned int spi_cpld_read(unsigned int offset)
 	data |= (((frame[1] >> 6) & 0xFFFF) << 16);
 	return data;
 }
+EXPORT_SYMBOL(spi_cpld_read);
 
 /*!
  * This function is called to write to a CPLD register over CSPI.
@@ -106,6 +107,7 @@ unsigned int spi_cpld_write(unsigned int offset, unsigned int reg_val)
 
 	return 0;
 }
+EXPORT_SYMBOL(spi_cpld_write);
 
 static int __devinit mx25_3ds_cpld_probe(struct spi_device *spi)
 {
diff --git a/arch/arm/mach-mx37/mx37_3stack_cpld.c b/arch/arm/mach-mx37/mx37_3stack_cpld.c
index dd9008a..ec99a22 100644
--- a/arch/arm/mach-mx37/mx37_3stack_cpld.c
+++ b/arch/arm/mach-mx37/mx37_3stack_cpld.c
@@ -74,6 +74,7 @@ unsigned int spi_cpld_read(unsigned int offset)
 	data |= (((frame[1] >> 6) & 0xFFFF) << 16);
 	return data;
 }
+EXPORT_SYMBOL(spi_cpld_read);
 
 /*!
  * This function is called to write to a CPLD register over CSPI.
@@ -106,6 +107,7 @@ unsigned int spi_cpld_write(unsigned int offset, unsigned int reg_val)
 
 	return 0;
 }
+EXPORT_SYMBOL(spi_cpld_write);
 
 static int __devinit mx37_3ds_cpld_probe(struct spi_device *spi)
 {
-- 
1.5.4.4

